png_benchmark.py

```diff
@@ -0,0 +1,152 @@
+                MIN_TIME_MS: 0,
+                AVG_TIME_MS: 0,
+                MAX_SIZE_B: 0,
+                MIN_SIZE_B: 0,
+                AVG_SIZE_B: 0,
+                MAX_RATIO_COMPRESSED_TO_ORIGINAL: 0,  # % of original size
+                MIN_RATIO_COMPRESSED_TO_ORIGINAL: 0,
+                AVG_RATIO_COMPRESSED_TO_ORIGINAL: 0,
+                FRAME_DATA: []
+            } for compress_level in range(INITIAL_COMPRESS_LEVEL, MAX_COMPRESS_LEVEL + 1)
+        } for compress_type in COMPRESS_TYPES
+    }
+
+    test_begin = time.time()
+    print("Start time:", test_begin)
+
+
+    for compress_type in COMPRESS_TYPES:
+        print(f"-----------------COMPRESS TYPE {compress_type}--------------------")
+        for compress_level in range(INITIAL_COMPRESS_LEVEL, MAX_COMPRESS_LEVEL + 1):
+            max_time_ns = 0
+            min_time_ns = float("inf")
+            total_time_ns = 0
+            max_size_B = 0
+            min_size_B = float("inf")
+            total_size_B = 0
+            max_compression_ratio = 0
+            min_compression_ratio = float("inf")
+            total_compression_ratio = 0
+            for frame_index in range(FRAME_COUNT):
+                img = Image.open(pathlib.Path(INPUT_PATH, f"{frame_index}.png"))
+                buffer = io.BytesIO()
+
+                # Running encode
+                gc.disable()
+                start = time.time_ns()
+                img.save(buffer, format="PNG", compress_level=compress_level, compress_type=compress_type)
+                end = time.time_ns()
+                gc.enable()
+
+                # Save singular test results
+                time_ns = end - start
+                size_B = buffer.getbuffer().nbytes
+                compression_ratio = size_B / os.path.getsize(pathlib.Path(INPUT_PATH, f"{frame_index}.png")) * 100
+
+                if time_ns > max_time_ns:
+                    max_time_ns = time_ns
+                elif time_ns < min_time_ns:
+                    min_time_ns = time_ns
+                
+                if size_B > max_size_B:
+                    max_size_B = size_B
+                elif size_B < min_size_B:
+                    min_size_B = size_B
+                
+                if compression_ratio > max_compression_ratio:
+                    max_compression_ratio = compression_ratio
+                elif compression_ratio < min_compression_ratio:
+                    min_compression_ratio = compression_ratio
+                
+                total_time_ns += time_ns
+                total_size_B += size_B
+                total_compression_ratio += compression_ratio
+                test_result = {
+                    "time_ns": time_ns,
+                    "size_B": size_B,
+                    "size_ratio_compressed_to_original_%": compression_ratio
+                }
+                results[f"compress_type_{compress_type}"][f"compress_level_{compress_level}"]["frame_data"].append(test_result)
```

Comment:
Line too long, maximum is 100 characters. Use intermediate string variables and/or intermediate dictionary.